### DISCUSSION TITLE
Rule engine

### DISCUSSION THREAD

**Discussion by ankurcts** [community/community#110890] [Actions]
### Select Topic Area

Bug

### Body

I have created a rule in the rule engine in the azure blob storage that is Validatealert and this rule defines the expression which contains the left and right side dynamic value but somehow, the right side value creates an issue. So are we supporting right side dynamic value or not? For example-
{
      'RuleName': 'ValidateAlert',
      'Expression':(account.AuthorizationKey.ToString()==(account.AuthorizationKey.ToString()==commonEvent.DynamicAuthorizationKey.ToString()),
'ErrorMessage': 'Failed : Dynamic alert is invalid.'
}

I'm getting the parser issue that is 'Exception while parsing expression Enum type 'DynamicAuthorizationKey' not found'. I tried with multiple permutation and combination to resolve that issue but just want to verify whether microsoft is supporting or not right side dynamic value in the rule engine definition.  

Kindly have a look into this issue, and please suggest to me an appropriate solution.

**Comment by notSaqlain** [comment]
While Azure Blob Storage Rule Engine allows using dynamic values on the left-hand side of expressions for comparison, it currently does not support dynamic values on the right-hand side. This means you cannot directly reference another dynamic value as part of the comparison.

This limitation is confirmed by the error message you're encountering, "Enum type 'DynamicAuthorizationKey' not found." The rule engine is unable to recognize commonEvent.DynamicAuthorizationKey.ToString() as a valid comparison operand because it expects a static value on the right-hand side.

Here are some alternative approaches you can consider to achieve your desired functionality:

**1. Pre-process dynamic values:**
Before defining the rule, retrieve the value of commonEvent.DynamicAuthorizationKey using C# code or another scripting language supported by Azure Functions.
Store this value in a temporary variable.
Use the temporary variable in your rule expression for comparison with account.AuthorizationKey.ToString().

**2. Use a different comparison operator:**
Instead of using the equality operator (==), consider using the Contains operator to check if the account.AuthorizationKey string contains the value retrieved from commonEvent.DynamicAuthorizationKey.
This approach might not be suitable if you need an exact match, but it can work if checking for the presence of a specific value within the authorization key is sufficient.

**3. Submit a feature request:**
If neither of the above solutions meets your specific needs, you can consider submitting a feature request to Microsoft Azure for enabling dynamic values on the right-hand side of rule engine expressions. You can use the Azure Feedback portal at https://feedback.azure.com/ to submit your request.

Remember that the chosen solution will depend on your specific requirements and the level of granularity needed in the comparison.

**[MARK AS COMPLETE IF THIS HELPS, THANKS!]**
